### PR TITLE
cdc: Support changing CDC configs dynamically (#10651)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #10651 to release-4.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tikv repo:
git pr https://github.com/tikv/tikv/pull/10684
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tikv.git pr/10684:release-4.0-d3761d610d3f
```

---

Signed-off-by: hi-rustin <rustin.liu@gmail.com>

### What problem does this PR solve?

Issue Number: close https://github.com/tikv/tikv/issues/10645

Problem Summary:

### What is changed and how it works?

What's Changed:

Support changing CDC configs dynamically.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch (Not Sure)

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

Side effects

None

### Release note 

```release-note
Support changing CDC configs dynamically
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
```
mysql> show config where type='tikv' and name like '%cdc%';
+------+-----------------+----------------------------------+--------+
| Type | Instance        | Name                             | Value  |
+------+-----------------+----------------------------------+--------+
| tikv | 127.0.0.1:20160 | cdc.incremental-scan-concurrency | 6      |
| tikv | 127.0.0.1:20160 | cdc.incremental-scan-speed-limit | 128MiB |
| tikv | 127.0.0.1:20160 | cdc.incremental-scan-threads     | 4      |
| tikv | 127.0.0.1:20160 | cdc.min-ts-interval              | 1s     |
| tikv | 127.0.0.1:20160 | cdc.sink-memory-quota            | 512MiB |
+------+-----------------+----------------------------------+--------+
5 rows in set, 1 warning (0.01 sec)

mysql> set config tikv `cdc.sink-memory-quota` = 10244444;
Query OK, 0 rows affected (0.03 sec)

mysql> show config where type='tikv' and name like '%cdc%';
+------+-----------------+----------------------------------+----------+
| Type | Instance        | Name                             | Value    |
+------+-----------------+----------------------------------+----------+
| tikv | 127.0.0.1:20160 | cdc.incremental-scan-concurrency | 6        |
| tikv | 127.0.0.1:20160 | cdc.incremental-scan-speed-limit | 128MiB   |
| tikv | 127.0.0.1:20160 | cdc.incremental-scan-threads     | 4        |
| tikv | 127.0.0.1:20160 | cdc.min-ts-interval              | 1s       |
| tikv | 127.0.0.1:20160 | cdc.sink-memory-quota            | 10244444 |
+------+-----------------+----------------------------------+----------+
```

**Review by overvenus — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

File: src/config.rs [hunk 1]
```diff
@@ -2703,6 +2705,7 @@ pub enum Module {
     Gc,
     Split,
     Unknown(String),
+    CDC,
 }
 
 impl From<&str> for Module {
```
[the remark this answers is not in the document]
Comment (hi-rustin) [reply src/config.rs:2711]: Added. Thanks for your review!

**Pull request merged**